plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java

```diff
@@ -318,15 +318,26 @@ private void warnExtraConfig(Path extraContentDir) {
         }
     }
 
-    private Path getDeploymentContent() {
+    private Path getDeploymentContent() throws MojoExecutionException {
         final PackageType packageType = PackageType.resolve(project);
         final String filename;
         if (this.filename == null) {
             filename = String.format("%s.%s", project.getBuild().getFinalName(), packageType.getFileExtension());
         } else {
             filename = this.filename;
         }
-        return targetDir.toPath().resolve(filename);
+        Path deployment = targetDir.toPath().resolve(filename);
+        if (Files.notExists(deployment)) {
+            if (this.filename != null ) {
+                throw new MojoExecutionException("No deployment found wih name " + this.filename);
+            }
+            if (this.runtimeName != null ) {
+                throw new MojoExecutionException("No deployment found although a runtime-name has been set. "
+                        + "Check the final name of the deployment. A custom file name can be set with <filename> parameter.");
```

Comment:
Can plugin log, what was he looking for? That mean to log filename.

Reply:
I will make small adjustement to the fix. Thank-you for your feedback.